lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -30,11 +30,19 @@ function Get-Vcredist {
         try {
             Write-JujuWarning "Trying to get vcredist Juju resource"
             $vcredistPath = Get-JujuResource -Resource "vcredist-x64"
-            return $vcredistPath
         } catch {
+            $vcredistPath = $null
             Write-JujuWarning "Failed downloading vcredist resource: $_"
-            Write-JujuWarning "Falling back to file download"
         }
+        if ($vcredistPath) {
+            $bytes = Get-Content $vcredistPath -TotalCount 31 -Encoding Byte
+            $str = [string]::join("", [char[]]$bytes)
+            if ($str -eq $DEFAULT_JUJU_RESOURCE_CONTENT) {
+                Throw "Manually attach the Juju resource 'vcredist-x64' or set the config option 'vcredist-url'."
+            }
+            return $vcredistPath
+        }
+        Write-JujuWarning "Falling back to file download"
```

Comment:
You're right. I missed something. The fallback will work if the user specifies the installer url for the vcredist installer. But, what won't work is taking a default download URL if the resource get doesn't work (aka people using Juju < 2.0 without any resources support).